api-tests/specs/practitioner-service/linkperson.js

```diff
@@ -0,0 +1,125 @@
+
+
+describe('As a user with Update Person permission I want to link a Person to an existing User So that they have the Persons roles when they login to the system', function () {
+
+    it('Given a Person exists who is not linked to a User AND a User exists that is not linked to a Person When I submit an API request to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are linked and I receive a success acknowledgement ', async () => {
+
+        //post a request to practitioner end point to create first Practitioner
+        const headers = await utils.getBootStrapUserHeadersWithAuth()
+        let fetchResponse = await fetch(conf.baseUrl + practitionerEndpointUri, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(requests.createFirstPerson),
+        })
+        let response = await fetchResponse.json()
+        firstPersonId = response.id
+
+        //link person to a user
+        let linkUserJSON = requests.linkUser;
+        linkpersonEndpointUri1 = conf.baseUrl + linkpersonEndpointUri.replace("{personId}", firstPersonId);
+
+        let fetchResponse1 = await fetch(linkpersonEndpointUri1, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(linkUserJSON),
+        })
+        expect(fetchResponse1.status).to.equal(HttpStatus.OK)
+
+    });
+
+    it('Given a Person exists who is not linked to a User When I submit an API request to link the Person to a User who is already linked to a Person in this trial Then the Person and User are NOT linked and I receive an error notification ', async () => {
+
+        //post a request to practitioner end point to create second Practitioner
+        const headers = await utils.getHeadersWithAuth()
+        let fetchResponse = await fetch(conf.baseUrl + practitionerEndpointUri, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(requests.createSecondPerson),
+        })
+        let response2 = await fetchResponse.json()
+        secondPersonId = response2.id;
+
+        //link person to a user who is already linked
+        let linkUserJSON = requests.linkUser;
+        linkpersonEndpointUri2 = conf.baseUrl + linkpersonEndpointUri.replace("{personId}", firstPersonId);
+        
+        //check application response confirms link failure
+        
+        let fetchResponse2 = await fetch(linkpersonEndpointUri2, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(linkUserJSON),
+        })
+        expect(fetchResponse2.status).to.equal(HttpStatus.UNPROCESSABLE_ENTITY)
+
+    });
+
+    it('When I submit an API request with missing Practitioner ID to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are NOT linked and I receive a failure acknowledgement ', async () => {
```

Comment:
would this be clearer?
Given a Person that is NOT linked to a User and a User who is not linked to a Person in this trial 

When I submit an API request with missing Practitioner ID to link the User and Person

Then the Person and User are NOT linked and I receive a failure acknowledgement

Reply:
Copied directly from the ARTS-185 story, the original link person story